src/ssz/tree_view/root.zig

```diff
@@ -0,0 +1,205 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const hashing = @import("hashing");
+const ListLengthUint = hashing.GindexUint;
+const Node = @import("persistent_merkle_tree").Node;
+const Gindex = @import("persistent_merkle_tree").Gindex;
+
+/// Represents the internal state of a tree view.
+///
+/// This struct manages the root node of the tree, caches child nodes and sub-data for efficient access,
+/// and tracks which child indices have been modified since the last commit.
+///
+/// It enables fast (re)access of children and batched updates to the merkle tree structure.
+pub const TreeViewData = struct {
+    root: Node.Id,
+
+    /// cached nodes for faster access of already-visited children
+    children_nodes: std.AutoHashMapUnmanaged(Gindex, Node.Id),
+
+    /// cached data for faster access of already-visited children
+    children_data: std.AutoHashMapUnmanaged(Gindex, TreeViewData),
+
+    /// whether the corresponding child node/data has changed since the last update of the root
+    changed: std.AutoArrayHashMapUnmanaged(Gindex, void),
+
+    /// Number of chunk nodes (starting from index 0) that have been prefetched into the cache.
+    /// Only used for array/list views with basic element types.
+    prefetched_chunk_count: ?usize,
+
+    /// Cached length for list views. null marks the value as invalid/unpopulated, so
+    /// the next `getLength()` fetches from the tree before returning a concrete length.
+    /// Only used for list views.
+    list_length: ?ListLengthUint,
```

Comment:
I don't think we want to track `list_length` for this common data structure
`list_length` should be tracked at its own list TreeView, `ListCompositeTreeView` for example
in `commit()` of `ListCompositeTreeView` we'll do an extra binding for `list_length`